tests/generator/base.py

```diff
@@ -269,31 +272,33 @@ def validate_generated_yaml(self, yaml_input):
         original (and normalized) input with the generated (and normalized)
         output.
         '''
-        output = '_generated_test_output.yaml'
-        output_path = os.path.join(self.confdir, output)
 
         for input in yaml_input:
-            lib.netplan_clear_netdefs()  # clear previous netdefs
-            lib.netplan_parse_yaml(input.encode(), None)
-            lib.write_netplan_conf_full(output.encode(), self.workdir.name.encode())
+            parser = netplan.Parser()
+            parser.load_yaml(input)
+            state = netplan.State()
+            state.import_parser_results(parser)
+
+            # TODO: Allow handling of full hierarchy overrides,
```

Comment:
I think this is fine for now, but we should create a task to keep track of it as it will not be possible to write tests that rely on merging without skipping this validation.

Reply:
:ballot_box_with_check:  FR-6338